/*
 * Copyright 2012-2016 Johns Hopkins University
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * Developed by: Roberts Group
 *               Johns Hopkins University
 *               http://biophysics.jhu.edu/roberts/
 *
 * Author(s): Elijah Roberts, Max Klein
 */
syntax = "proto2";
package lm.input;

import "lm/types/BoundaryConditions.proto";
import "lm/types/Lattice.proto";

message DiffusionModel {
  required int32 number_species                             = 1;
  required int32 number_reactions                           = 2;
  required int32 number_site_types                          = 3;
  repeated double diffusion_matrix                          = 4 [packed=true];
  repeated uint32 reaction_location_matrix                  = 5 [packed=true];
  required double lattice_spacing                           = 6;
  required lm.types.Lattice initial_lattice                 = 7;
  optional lm.types.BoundaryConditions boundary_conditions  = 8;
}
